utils/validations/test_validate_sarif.go

```diff
@@ -0,0 +1,242 @@
+package validations
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/jfrog/jfrog-cli-security/utils/formats/sarifutils"
+	"github.com/jfrog/jfrog-cli-security/utils/jasutils"
+	"github.com/jfrog/jfrog-cli-security/utils/results/conversion/sarifparser"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+	"github.com/stretchr/testify/assert"
+)
+
+// Content should be a Json string of sarif.Report and will be unmarshal.
+// Value is set as the Actual content in the validation params
+func VerifySarifResults(t *testing.T, content string, params ValidationParams) {
+	var results sarif.Report
+	err := json.Unmarshal([]byte(content), &results)
+	assert.NoError(t, err)
+	params.Actual = results
+	ValidateCommandSarifOutput(t, params)
+}
+
+func ValidateCommandSarifOutput(t *testing.T, params ValidationParams) {
+	results, ok := params.Actual.(*sarif.Report)
+	if assert.True(t, ok) {
+		ValidateSarifIssuesCount(t, params, results)
+		if params.Expected != nil {
+			expectedResults, ok := params.Expected.(*sarif.Report)
+			if assert.True(t, ok) {
+				ValidateSarifResults(t, params.ExactResultsMatch, expectedResults, results)
+			}
+		}
+	}
+}
+
+func ValidateSarifIssuesCount(t *testing.T, params ValidationParams, results *sarif.Report) {
+	var vulnerabilities, securityViolations, licenseViolations, applicableResults, undeterminedResults, notCoveredResults, notApplicableResults int
+
+	iac := sarifutils.GetResultsLocationCount(sarifutils.GetRunsByToolName(results, sarifparser.IacToolName)...)
+	secrets := sarifutils.GetResultsLocationCount(sarifutils.GetRunsByToolName(results, sarifparser.SecretsToolName)...)
+	sast := sarifutils.GetResultsLocationCount(sarifutils.GetRunsByToolName(results, sarifparser.SastToolName)...)
+
+	scaRuns := sarifutils.GetRunsByToolName(results, sarifparser.ScaToolName)
+	for _, run := range scaRuns {
+		for _, result := range run.Results {
+			// If watch property exists, add to security violations or license violations else add to vulnerabilities
```

Comment:
what about operational risks? don't we show it at all?

Reply:
Not supported in SARIF format yet, we can add as a feature to the format.

Reply:
wdym we can add?